measurements/attribute_based.py

```diff
@@ -62,56 +62,59 @@ def size_and_distance(dataloader, args):
+                if FACE_DETECT == 0:
+                    detect_info = detect_face(target[3], detect_info, client)
+                    faceDetails = detect_info[target[3]]['FaceDetails']
+                    if len(detect_info) % 20 == 0:
+                        pickle.dump(detect_info, open('{}_rekognitioninfo.pkl'.format(args.folder), 'wb'))
 
-                yes_face = False
-                for face in faceDetails:
-                    if face['Confidence'] > .9:
+                    yes_face = False
+                    for face in faceDetails:
+                        if face['Confidence'] > .9:
+                            yes_face = True
+                elif FACE_DETECT == 1:
+                    image = cv2.imread(target[3])
+                    gray = cv2.cvtColor(image, cv2.COLOR_RGB2GRAY)
+                    faces = faceCascade.detectMultiScale(
+                    gray,
+                    scaleFactor=1.1,
+                    minNeighbors=5,
+                    #minSize=(30, 30),
+                    flags = cv2.CASCADE_SCALE_IMAGE
+                    )
+                    yes_face = False
+                    if len(faces) > 0:
                         yes_face = True
-            elif FACE_DETECT == 1:
-                image = cv2.imread(target[3])
-                gray = cv2.cvtColor(image, cv2.COLOR_RGB2GRAY)
-                faces = faceCascade.detectMultiScale(
-                   gray,
-                   scaleFactor=1.1,
-                   minNeighbors=5,
-                   #minSize=(30, 30),
-                   flags = cv2.CASCADE_SCALE_IMAGE
-                )
-                yes_face = False
-                if len(faces) > 0:
-                    yes_face = True
-
-            # If there's no face detected and the person is too small, attribute cannot be distinguished
-            if not yes_face or pixel_size < 1000.:
-                scene_group = target[4]
-                if not yes_face:
-                    no_faces[attribute[0]].append((size, pixel_size, scene_group))
-                elif pixel_size < 1000.:
-                    tiny_sizes[attribute[0]].append((size, scene_group))
-                continue
 
-            sizes[attribute[0]].append(size)
-            distances[attribute[0]].append(distance)
+                # If there's no face detected and the person is too small, attribute cannot be distinguished
+                if not yes_face or pixel_size < 1000.:
+                    scene_group = target[4]
+                    if not yes_face:
+                        for att in attribute[0]:
+                            no_faces[att].append((size, pixel_size, scene_group))
+                    elif pixel_size < 1000.:        
+                        for att in attribute[0]:
+                            tiny_sizes[att].append((size, scene_group))
+                    continue
+                for att in attribute[0]:
```

Comment:
according to this logic, for every bbox, you would be adding the size to every single attribute in this image? I think you would only want to add this one size to the singular attribute that this particular bbox corresponds to